File: HackerRank/plusMinus/plusMinus.py
#!/bin/python3


def getInput():
    int(input().strip())
    inputStr  = input().strip()
    return [int(s) for s in inputStr.split()]


def main():
    POS = NEG = ZERO = 0
    for number in getInput():
        if number > 0:
            POS += 1
        elif number == 0:
            ZERO += 1
        else:
            NEG += 1
    print(POS, NEG, ZERO)


if __name__ == '__main__':
    main()

File: HackerRank/plusMinus/plusMinus.go
package plusMinus

import (
	"log"
	"os"
	"strconv"
	"strings"
)

func get_input() []string {
	cli := &input.UI{
		Writer: os.Stdout,
		Reader: os.Stdin,
	}
	_, err := cli.Ask("", &input.Options{
		Required: true,
		Loop:     true,
	})
	if err != nil {
		log.Fatal(err)
	}
	inStr, err := cli.Ask("all numbers space separated", &input.Options{
		Required: true,
		Loop:     true,
	})
	if err != nil {
		log.Fatal(err)
	}
	return strings.Split(inStr, " ")
}
func main() {
	POS, NEG, ZERO := 0, 0, 0
	for _, numberString := range get_input() {
		number, err := strconv.Atoi(numberString)
		if err != nil {
			log.Fatal(err)
		}
		switch {
		case number < 0:
			POS++
		case number > 0:
			NEG++
		default:
			ZERO++
		}
	}
	println(POS, " ", NEG, " ", ZERO)
}
